src/compiler/checker.ts

```diff
@@ -13548,6 +13548,20 @@ namespace ts {
             return _jsxElementChildrenPropertyName;
         }
 
+        function getApparentTypeOfJsxPropsType(propsType: Type): Type {
+            if (propsType) {
+                if (propsType.flags & TypeFlags.Intersection) {
+                    const propsApprentType: Type[] = [];
+                    for (const t of (<UnionOrIntersectionType>propsType).types) {

```

Comment:
Later: no it doesn't. :( You should be able to express this as `getIntersectionType(t.types.map(getApparentType)))` though, right?